src/which_within_distance.cpp

```diff
@@ -0,0 +1,28 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// [[Rcpp::export]]
+IntegerVector which_within_dist(NumericMatrix distmat, NumericVector idx, double dist) {
+
+  int n_idx = idx.size();
+  int n_matrix = distmat.ncol();
+
+  int n_pos = 0;
+  for (int i = 0; i < n_idx; i++) {
+    n_pos += sum(distmat(idx[i] - 1, _) <= dist);
+  }
+
+  int p = 0;
+  IntegerVector out(n_pos);
+  for (int i = 0; i < n_idx; i++) {
```

Comment:
Hee I would suggest that you moving defining `int id = idx[i] - 1` instead of doing it inside the second loop.